### PR TITLE
Add new event types to application server

### PR DESCRIPTION
**Pull request by vlasebian**
#### Summary

References [#4227](https://github.com/TheThingsIndustries/lorawan-stack/issues/4227) (tti issue).

#### Changes

Add events that can be associated with event types propagated by webhooks.

##### Steps

1. Have a local instance of the stack up and running with an app and a device
2. Simulate uplinks for each type of message that triggers a specific event, e.g.

```bash
APP=$1
DEV=$2

curl -i "http://localhost:1885/api/v3/as/applications/$APP/devices/$DEV/up/simulate" \
  -H "Authorization: $TTS_AUTH_TOKEN" \
  -d '{
  "downlink_ack": {
    "session_key_id": "string",
    "f_port": 0,
    "f_cnt": 0,
    "frm_payload": "string",
    "decoded_payload": {},
    "decoded_payload_warnings": [
      "string"
    ],
    "confirmed": true,
    "class_b_c": {
      "gateways": [
        {
          "gateway_ids": {
            "gateway_id": "string",
            "eui": "70B3D57ED000ABCD"
          },
          "antenna_index": 0,
          "group_index": 0
        }
      ],
      "absolute_time": "2024-08-27T08:50:56.236Z"
    },
    "priority": "LOWEST",
    "correlation_ids": [
      "string"
    ],
    "confirmed_retry": {
      "attempt": 1,
      "max_attempts": 100
    }
  }
}'
```
3. Check the live data tab in the console

##### Results

Screenshot 2024-08-27 at 15 55 35

##### Regressions

None.

#### Notes for Reviewers

None.

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [x] Documentation: Relevant documentation is added or updated.
- [x] Testing: The steps/process to test this feature are clearly explained including testing for regressions.
- [x] Infrastructure: If infrastructural changes (e.g., new RPC, configuration) are needed, a separate issue is created in the infrastructural repositories.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/applicationserver/observability.go [hunk 1]
```diff
@@ -101,6 +101,36 @@ var (
 		events.WithDataType(&ttnpb.ApplicationUp{}),
 		events.WithPropagateToParent(),
 	)
+	evtForwardDownlinkQueueInvalidated = events.Define(
+		"as.up.downlink.queue.invalidated.forward", "forward downlink queue invalidated",
```
Comment (johanstokking) [inline pkg/applicationserver/observability.go]: I know it's confusing, but this should fall under the `as.down.data` hierarchy, like below.

**Review by johanstokking — CHANGES_REQUESTED** [review]
Otherwise looks good

File: pkg/applicationserver/observability.go (hunk 1 above)
Comment (vlasebian) [reply pkg/applicationserver/observability.go]: Fixed!

**Review by johanstokking — APPROVED** [review]
Great, thanks.

**Review by KrishnaIyer — APPROVED** [review]
Nice. LGTM

**Comment by KrishnaIyer** [comment]
Please run `tools/bin/mage go:messages` to resolve the diff.

**Pull request merged**